convert2rhel/actions/__init__.py

```diff
@@ -206,25 +211,94 @@ def run(self):
 
         self._has_run = True
 
-    def set_result(self, status=_NO_USER_VALUE, error_id=_NO_USER_VALUE, message=_NO_USER_VALUE):
+    @property
+    def result(self):
+        return self._result
+
+    @result.setter
+    def result(self, action_message):
+        """
+        Make sure the correct policy for result messages is used.
+        """
+        if not isinstance(action_message, ActionResult):
+            raise Exception()
+
+        self._result = action_message
+
+    def set_result(self, level, id=None, message=None):
+        """
+        Helper method that sets the resulting values for level, id and message.
+
+        :param id: The id to identify the result.
+        :type id: str
+        :param level: The status_code of the result .
+        :type: level: str
+        :param message: The message to be set.
+        :type message: str | None
+        """
+        if level not in ("ERROR", "OVERRIDABLE", "SKIP", "SUCCESS"):
+            raise KeyError("The level of result must be FAILURE, OVERRIDABLE, SKIP, or SUCCESS.")
+
+        result = ActionResult(level, id, message)
+        self.result = result
```

Comment:
Not a necessary change, but I would lean towards doing something like: 


```suggestion
        self.result = ActionResult(level, id, message)
```

Personally, I don't like setting the result (or instance, in that case) of something to a variable and then doing an assigment adds too much value to the code, instead, on cases like this, I would assign it directly to the thing we want to assign.

Reply:
I'll add this